lib/compat/wordpress-6.3/blocks.php

```diff
@@ -120,3 +120,50 @@ function gutenberg_wp_block_register_post_meta() {
 	);
 }
 add_action( 'init', 'gutenberg_wp_block_register_post_meta' );
+
+/**
+ * Allow querying blocks by sync_status.
+ *
+ * Note: This should be removed when the minimum required WP version is >= 6.3.
+ *
+ * @param array           $args    Array of arguments for WP_Query.
+ * @param WP_REST_Request $request The REST API request.
+ *
+ * @return array Updated array of arguments for WP_Query.
+ */
+function gutenberg_rest_wp_block_query( $args, $request ) {
```

Comment:
This patch adds the ability to pass `sync_status` query parameter to the `/wp/v2/blocks` api. Allowed values for now are `fully` and `unsynced`.